src/pages/workspaces/workspace/Dashboard.js

```diff
@@ -420,6 +427,18 @@ const WorkspaceDashboard = _.flow(
     ]
   }
 
+  const updateRecentlyViewedWorkspaces = () => {
```

Comment:
I think it would be better to put this in `WorkspaceContainer` or `wrapWorkspace` in WorkspaceContainer.js. That way, recently viewed workspaces would be updated if the user navigates directly to some other workspace page like Data or Notebooks instead of the dashboard.